pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -338,8 +303,12 @@ func (e *EndpointController) syncService(key string) error {
 		glog.V(4).Infof("Finished syncing service %q endpoints. (%v)", key, time.Now().Sub(startTime))
 	}()
 
-	obj, exists, err := e.serviceStore.Indexer.GetByKey(key)
-	if err != nil || !exists {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	service, err := e.serviceLister.Services(namespace).Get(name)
```

Comment:
how hard is it to deep copy the map copies that happening?  There's no mutation on `newEndpoints.Labels` as a for instance, but the map is the same as the shared cache for services.

If its non-trivial (it always is in Go), I could live without it.